backend/src/v4/models/helper/model.js

```diff
@@ -299,39 +299,24 @@ function searchTree(account, model, branch, rev, searchString, username) {
 
 }
 
-function listSubModels(account, model, branch = "master") {
-
+async function listSubModels(account, model) {
+	// This method is in the hot-path for validating permissions
+	const containers = await getSubModels(account, model);
 	const subModels = [];
-
-	return History.findByBranch(account, model, branch).then(history => {
-
-		if(history) {
-			return getRefNodes(account, model, branch);
-		} else {
-			return [];
-		}
-
-	}).then(refs => {
-
-		const proms = refs.map(ref =>
-
-			findModelSettingById(ref.owner, ref.project, { name: 1 }).then(subModel => {
-				// TODO: Why would this return null?
-				if (subModel) {
-					subModels.push({
-						database: ref.owner,
-						model: ref.project,
-						name: subModel.name
-					});
-				}
-
-			})
-
-		);
-
-		return Promise.all(proms).then(() => Promise.resolve(subModels));
-
+	const proms = containers.map((container) => {
+		findModelSettingById(account, container.model, { name: 1 }).then(subModel => {
+			// TODO: Why would this return null?
```

Comment:
I guess if for some reason the federation has an entry to a container that no longer exists? We don't anticipate this to happen in normal situatiosn for sure